bumble/gatt_client.py

```diff
@@ -548,6 +551,23 @@ async def subscribe(self, characteristic, subscriber=None):
 
         await self.write_value(cccd, struct.pack('<H', bits), with_response=True)
 
+    async def unsubscribe(self, characteristic):
+        # If we haven't already discovered the descriptors for this characteristic, do it now
+        if not characteristic.descriptors_discovered:
+            await self.discover_descriptors(characteristic)
+
+        # Look for the CCCD descriptor
+        cccd = characteristic.get_descriptor(GATT_CLIENT_CHARACTERISTIC_CONFIGURATION_DESCRIPTOR)
+        if not cccd:
+            logger.warning('unsubscribing to characteristic with no CCCD descriptor')
+            return
+
+        # This removes all subscribers for this attribute from the sets!
+        self.notification_subscribers.pop(characteristic.handle, None)
+        self.indication_subscribers.pop(characteristic.handle, None)
```

Comment:
It seems useful to have an unsubscribe method.

However, I'm not sure how to handle removing subscribers. Should we remove a specific subscriber that the same caller previously added with subscribe()? How can we figure out which subscriber that is? It's a common pattern to use a lambda function for the subscriber, which means the caller can't go back and get a reference to it.

Since we're setting the CCCD bits to 0, none of the subscribers are going to get any updates for this characteristic. So for now, I'm removing all of them.

Reply:
I'd suggest adding a `subscriber` parameter to `unsubscribe`, which would default to `None`. If not `None`, this would just remove that specific subscriber. If `None` it would remove all subscribers. That seems pretty simple. It doesn't cover all cases, because you couldn't selectively remove the lambda subscriber that gets added automatically (without removing the explicit subscribers as well), but I don't think that's a problem.

Reply:
Sounds good. Removing a matching subscriber if provided. Otherwise removing all subscribers (and lambdas) for attribute.

Added some unsubscribe asserts to tests/gatt_test.py to check logic.